### ISSUE TITLE
Saga not firing from connected action

### ISSUE THREAD

**Issue by prsauer** [keajs/kea#84]
I have a logic (logic A) that connects to a reducer and an action in a separate logic (logic B)

Logic B has a saga attached to this action on takeEvery

Logic A fires the action as a part of one of its own actions, which properly triggers the reducer but doesn't trigger the saga from takeEvery in logic B.

Is this expected behavior? How do I organize my code such that I can connect to logic B and have the sagas fire?

Similar to #81 although I'm using saga which was the suggestion I can't seem to get it to work

**Comment by prsauer** [comment]
Root cause of this issue was actually unrelated to connect() semantics

I'm using
https://github.com/wix/react-native-navigation

and on a transition from component A to component B it calls A.willMount before B.willUnmount which causes saga to break if both A and B depend on the same logic

Is there a fix for this? Can I keep my sagas connected independent of life cycle calls?

**Comment by mikaoelitiana** [comment]
@prsauer did you find a way to make this work? I am currently having the same issue.

**Comment by prsauer** [comment]
@mikaoelitiana Kind of. You can take a look at my fork:
https://github.com/prsauer/kea-saga

I basically removed the mechanism that disconnects the sagas when the components unmount. It's not very elegant or clean but it does work. Definitely need to make sure it's appropriate for your use case.

**Comment by k1sul1** [comment]
I had a similar issue. We have "multiple stores", as in multiple files with `export default kea({})` in them, different files for different things. There's a load of things that have to happen before anything else happens in the app, so let's say I have A, B and C. 

A has an initialize action, that is called in componentDidMount of the root component of our app. Said action triggers a saga, that runs actions from both B and C.

```javascript
import b from 'b'
import c from 'c'
// kea({
  takeLatest: ({ actions, workers }) => ({
    [actions.initialize]: function * ({ payload }) {
      yield put(b.actions.initialize())
      yield put(b.actions.initialize())
      yield put(this.actions.setReady(true))
    },
   }),
// })
```
B and C also have initialize actions, that run similar sagas, but their own things of course. It took me ages to figure out why B completed succesfully, but C never even started. I added `console.log`s to the actions, and could see that both B and C logged whatever I put them to log successfully.  

In B saga, I could `console.log` successfully. In C, nothing happened. 

After contemplating my life for about 4 hours, I used rubber :duck:  debugging and said my coworker the following

> It can't be that I haven't connected it to anything, right? 

and went on to try a single thing before going home.

In my App.js, I had 

```javascript
@connect({
  actions: [
    application, [
      'setError',
      'initialize',
    ],
  ],
  props: [
    b, [
      'bthing'
    ],
  ],
})
export default class App extends Component {
```

I changed it to 
```javascript
@connect({
  actions: [
    application, [
      'setError',
      'initialize',
    ],
  ],
  props: [
    b, [
      'bthing'
    ],
    c, [
      'cthing'
    ],
  ],
})
export default class App extends Component {
```

and proceeded to flip the table, because it fucking worked. It must be because if the store isn't connected to any component, this never runs, and firing actions from other stores doesn't work.

```javascript
  start: function * () {
    while (true) {
      const action = yield take(responseChannel)
      yield put(action)
    }
  },

```

Let's hope this helps someone. Or me, when this happens to me again in about six months.

**Comment by prsauer** [comment]
@k1sul1 yeah I had a similar issue earlier in my debugging -- if the component is never mounted then the sagas will never get created and effects tied to them won't happen. 

It's a consequence of how kea-saga is implemented which needs a big overhaul to get rid of easy to write errors like these. Not sure where the maintainer has been, my fork doesn't directly address your issue.

**Comment by k1sul1** [comment]
It's a little rough around some edges but it's 4 months in and I wouldn't go back to "vanilla" redux anymore. 

I think things would be at least a bit better if there were warnings on some things. It's easy to make a typo when defining a reducer, and then wonder why your action won't do anything.

**Comment by benjamin-glitsos** [comment]
Hey @mariusandra , I've been trying and failing to make kea-saga (or kea-thunk) work for a few hours so far. It's just 'not firing'. Would you know of any example projects with kea-saga working on Create React App?

**Comment by mariusandra** [comment]
Hey @benjamin-glitsos , I assume what's happening is that you're passing the `sagaPlugin` to `getStore`? If so, please try running something like this instead:

```js
import { activatePlugin } from 'kea'
import sagaPlugin from 'kea-saga'
activatePlugin(sagaPlugin)
```

... somewhere near the beginning of `src/store.js` or `src/index.js`, before `<App />` is imported.

**Comment by benjamin-glitsos** [comment]
Omg you genius! It works :D 
I cant believe you got back to me so quickly. This is a rly beautiful library by the way, it should become really widely used in my opinion.
